airflow/jobs/scheduler_job.py

```diff
@@ -510,7 +510,15 @@ def _process_executor_events(self, session: Session = None) -> int:
 
         # Check state of finished tasks
         filter_for_tis = TI.filter_for_tis(tis_with_right_state)
-        tis: List[TI] = session.query(TI).filter(filter_for_tis).options(selectinload('dag_model')).all()
+        query = session.query(TI).filter(filter_for_tis).options(selectinload('dag_model'))
+        # row lock this entire set of tasks to make sure the scheduler doesn't fail when we have
+        # multi-schedulers
+        tis: List[TI] = with_row_locks(
+            query,
+            of=TI,
+            session=session,
+            **skip_locked(session=session),
+        ).all()
```

Comment:
```suggestion
        tis: Iterator[TI] = with_row_locks(
            query,
            of=TI,
            session=session,
            **skip_locked(session=session),
        )
```

Probably a good time to change this

Reply:
Actually yeah - now when we do "for update" for those rows, changing to iterator makes sense.